src/org/zaproxy/zap/extension/websocket/ui/WebSocketMessagesPayloadFilter.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.websocket.ui;
+
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+
+import java.util.regex.Pattern;
+
+/**
+ * Value object of the payload filter for the {@link WebSocketMessagesView}. Used as filter for the {@link WebSocketPanel} restricting payloads of messages
+ * shown in WebSockets tab.
+ */
+public class WebSocketMessagesPayloadFilter {
+	private String stringPayloadPattern;
+	/**
+	 * Defines the expression used to filter the websocket messages in the
+	 * {@link WebSocketMessagesView} by their payload.
+	 * This expression is either an regular expression or a simple search
+	 * string.
+	 */
+	private Pattern payloadPattern;
+	
+	/**
+	 * Determines if the expression of the filter is a regular expression. 
+	 */
+	private Boolean isRegularExpression;
+	
+	/**
+	 * Determines if the filter should be case sensitive.
+	 */
+	private Boolean isIgnoreCase;
+
+	/**
+	 * Determines if the matches of the filter should be inverted.
+	 */
+	private Boolean isInverted;
+
+	public Pattern getPayloadPattern() {
+		return payloadPattern;
+	}
+
+	public void setPattern(Pattern pattern) {
+		this.payloadPattern = pattern;
+	}
+
+	public Boolean getRegularExpression() {
+		return isRegularExpression;
+	}
+
+	public Boolean getIgnoreCase() {
+		return isIgnoreCase;
+	}
+
+
+	public Boolean getInverted() {
+		return isInverted;
+	}
+
+	public void setInverted(Boolean inverted) {
+		isInverted = inverted;
+	}
+
+	public String getStringPayloadPattern() {
+		return stringPayloadPattern;
+	}
+
+	public String getQueryString(){
+		StringBuilder sb = new StringBuilder(stringPayloadPattern);
+		sb.insert(0,'%');
+		sb.append('%');
+		return sb.toString();
+	}
+
+	public void compilePattern(){
+		if(!isRegularExpression){
+			 stringPayloadPattern = escapeMetaCharacters(stringPayloadPattern);
```

Comment:
`Pattern.quote`, note that we shouldn't change `stringPayloadPattern` itself, this should be used only when compiling it otherwise in the UI we show something different than what was specified by the user.